lib/amqp.js

```diff
@@ -19,9 +20,54 @@ class Amqp {
         const snapshotRateLimit = this.settings && this.settings.SNAPSHOT_RATE_LIMIT
             ? this.settings.SNAPSHOT_RATE_LIMIT : 20;
 
-        this.sendData = pThrottle(this.sendDataUnlimited, dataRateLimit, 1000);
-        this.sendError = pThrottle(this.sendErrorUnlimited, errorRateLimit, 1000);
-        this.sendSnapshot = pThrottle(this.sendSnapshotUnlimited, snapshotRateLimit, 1000);
+        this.asyncSendData = pThrottle(this.sendDataUnlimited, dataRateLimit, 1000);
+        this.asyncSendError = pThrottle(this.sendErrorUnlimited, errorRateLimit, 1000);
+        this.asyncSendSnapshot = pThrottle(this.sendSnapshotUnlimited, snapshotRateLimit, 1000);
+    }
+
+    sendData(data, properties) {
+        let done = false;
+        let result = null;
+        let promise = this.asyncSendData(data, properties);
+        promise.then((value) => {
```

Comment:
This is a seriously wrong approach. Please speak to the node.js experts.

Reply:
I know, but the problem is that all function call rate limitrors that I tested and which are good to use (p-throttle and bottleneck) are produsing async functions. So correct option would be to refactor sailor totaly using async/await function calling (also tests). But this will take really mush time, so I need to synchronize this functions call in some way... Ok, I'll try to find better way to do it.

Reply:
@ElBabbler I understood your problem, it's indeed a bit tricker than we initially expected. Just give me ~1 hour, I'll prepare some ideas on it.